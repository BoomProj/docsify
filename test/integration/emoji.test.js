const docsifyInit = require('../helpers/docsify-init');

// Suite
// -----------------------------------------------------------------------------
describe('Emoji', function () {
  // Tests
  // ---------------------------------------------------------------------------
  const sharedConfig = {
    markdown: {
      homepage: `
        hh:mm

        hh:mm:

        hh:mm:ss

        :mm:s

        Namespace::SubNameSpace

        Namespace::SubNameSpace::Class

        2014-12-29T16:11:20+00:00

        2020-12-03T15:05:57+00:00 (extended)

        2020-12-03T15:05:57Z (extended)

        :01:

        :smile:

        1:smile:

        :smile:1

        1:smile:1

        :smile::smile::smile:

        :smile: :smile: :smile:
      `,
    },
  };

  test('Native emoji', async () => {
    await docsifyInit({
      config: {
        nativeEmoji: true,
      },
      ...sharedConfig,
      // _logHTML: true,
    });

    const mainElm = document.querySelector('#main');

    expect(mainElm.innerHTML).toMatchSnapshot();
  });

  test('Fallback emoji (images)', async () => {
    await docsifyInit({
      config: {
        nativeEmoji: false,
      },
      ...sharedConfig,
      // _logHTML: true,
    });

    const mainElm = document.querySelector('#main');

    expect(mainElm.innerHTML).toMatchSnapshot();
  });

  test('No emoji', async () => {
    await docsifyInit({
      config: {
        noEmoji: true,
      },
      ...sharedConfig,
      // _logHTML: true,
    });

    const mainElm = document.querySelector('#main');

    expect(mainElm.innerHTML).toMatchSnapshot();
  });
});
