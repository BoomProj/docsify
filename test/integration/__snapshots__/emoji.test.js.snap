// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji Fallback emoji (images) 1`] = `"<p>hh:mm</p><p>hh:mm:</p><p>hh:mm:ss</p><p>:mm:s</p><p>Namespace::SubNameSpace</p><p>Namespace::SubNameSpace::Class</p><p>2014-12-29T16:11:20+00:00</p><p>2020-12-03T15:05:57+00:00 (extended)</p><p>2020-12-03T15:05:57Z (extended)</p><p>:01:</p><p><img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\"></p><p>1<img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\"></p><p><img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\">1</p><p>1<img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\">1</p><p><img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\"><img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\"><img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\"></p><p><img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\"> <img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\"> <img class=\\"emoji\\" src=\\"https://github.githubassets.com/images/icons/emoji/unicode/1f604.png\\" alt=\\"1\\"></p>"`;

exports[`Emoji Native emoji 1`] = `"<p>hh:mm</p><p>hh:mm:</p><p>hh:mm:ss</p><p>:mm:s</p><p>Namespace::SubNameSpace</p><p>Namespace::SubNameSpace::Class</p><p>2014-12-29T16:11:20+00:00</p><p>2020-12-03T15:05:57+00:00 (extended)</p><p>2020-12-03T15:05:57Z (extended)</p><p>:01:</p><p><span class=\\"emoji\\">ðŸ˜„ï¸Ž</span></p><p>1<span class=\\"emoji\\">ðŸ˜„ï¸Ž</span></p><p><span class=\\"emoji\\">ðŸ˜„ï¸Ž</span>1</p><p>1<span class=\\"emoji\\">ðŸ˜„ï¸Ž</span>1</p><p><span class=\\"emoji\\">ðŸ˜„ï¸Ž</span><span class=\\"emoji\\">ðŸ˜„ï¸Ž</span><span class=\\"emoji\\">ðŸ˜„ï¸Ž</span></p><p><span class=\\"emoji\\">ðŸ˜„ï¸Ž</span> <span class=\\"emoji\\">ðŸ˜„ï¸Ž</span> <span class=\\"emoji\\">ðŸ˜„ï¸Ž</span></p>"`;

exports[`Emoji No emoji 1`] = `"<p>hh:mm</p><p>hh:mm:</p><p>hh:mm:ss</p><p>:mm:s</p><p>Namespace::SubNameSpace</p><p>Namespace::SubNameSpace::Class</p><p>2014-12-29T16:11:20+00:00</p><p>2020-12-03T15:05:57+00:00 (extended)</p><p>2020-12-03T15:05:57Z (extended)</p><p>:01:</p><p>:smile:</p><p>1:smile:</p><p>:smile:1</p><p>1:smile:1</p><p>:smile::smile::smile:</p><p>:smile: :smile: :smile:</p>"`;
